common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -199,11 +205,22 @@ protected void serviceInit(Configuration externalConf) throws Exception {
     _conf = new Configuration(externalConf);
     URL confOverlayUrl = Thread.currentThread().getContextClassLoader()
         .getResource(SHUFFLE_SERVICE_CONF_OVERLAY_RESOURCE_NAME);
+    String className = YarnShuffleService.class.getName();
     if (confOverlayUrl != null) {
       logger.info("Initializing Spark YARN shuffle service with configuration overlay from {}",
           confOverlayUrl);
       _conf.addResource(confOverlayUrl);
     }
+
+    String logsNamespace = _conf.get(SPARK_SHUFFLE_SERVICE_LOGS_NAMESPACE_KEY, "");
+    if (!logsNamespace.isEmpty()) {
+      logger = LoggerFactory.getLogger(className + "." + logsNamespace);
+    }
+    if (logsNamespace.contains(" ")) {
+      logger.warn("Logging namespace contains a space which might " +
+              "not be parseable by some frameworks");
```

Comment:
I think we can make the message a bit more actionable by including the config name (`SPARK_SHUFFLE_SERVICE_LOGS_NAMESPACE_KEY`) and explaining the suggested format ("... It is suggested to configure this to a simple string that looks like a Java package or class name to conform to standard logger name format")

Reply:
Actually while thinking about this further, maybe it doesn't make sense to log a warning. Users may not be leveraging any framework which expects logger names to look like class names, and then they will get an unnecessary warning. Perhaps we can just rely on the note in the config documentation which says to be careful about using spaces. @mridulm what do you think?